docs(logger): document enriching logs with logrecord attributes (#1271)
awslabs/aws-lambda-powertools-python: docs/core/logger.md

---8<-- "examples/logger/src/cloning_logger_config.py"
```

**How can I add standard library logging attributes to a log record?**

> The Python standard library log records contains a [large set of atttributes](https://docs.python.org/3/library/logging.html#logrecord-attributes), however only a few are included in the powertools log record by default.

Reviewer: ```suggestion
The Python standard library log records contains a [large set of atttributes](https://docs.python.org/3/library/logging.html#logrecord-attributes){target="_blank"}, however only a few are included in Powertools Logger log record by default.
```

> If you need to add additional records, these can be included as `kwargs` to `Logger`, or to `LambdaPowertoolsFormatter` when they are being instantiated, or to managed later with the `append_keys` or `remove_keys` methods.

Reviewer: ```suggestion
You can include any of these logging attributes as key value arguments (`kwargs`) when instantiating `Logger` or `LambdaPowertoolsFormatter`.

You can also add them later anywhere in your code with `append_keys`, or remove them with `remove_keys` methods.
```

Broke line length to ease reading while keeping the same message.

=== "collect.py"

    ```python hl_lines="3 8 10"
    ---8<-- "examples/logger/src/append_and_remove_keys.py"
    ```
=== "Example CloudWatch Logs excerpt"

    ```json hl_lines="6 15-16"
    ---8<-- "examples/logger/src/append_and_remove_keys.json"
    ```

> For log records originating from powertools `Logger`, the `name` attribute will be the same as `service`, for log records coming from standard library logger, it will be the name of the logger (i.e. what was used as name argument to `logging.getLogger`).

Reviewer: ```suggestion
For log records originating from Powertools `Logger`, the `name` attribute will be the same as `service`, for log records coming from standard library logger, it will be the name of the logger (i.e. what was used as name argument to `logging.getLogger`).
```

---8<-- "examples/logger/src/cloning_logger_config.py"
```

### How can I add standard library logging attributes to a log record?

> The Python standard library log records contains a [large set of atttributes](https://docs.python.org/3/library/logging.html#logrecord-attributes){target="_blank"}, however only a few are included in Powertools Logger log record by default.

Author: Nit, but.... "Powertools `Logger`" instead of `Powertools Logger"? 

```suggestion
The Python standard library log records contains a [large set of atttributes](https://docs.python.org/3/library/logging.html#logrecord-attributes){target="_blank"}, however only a few are included in Powertools `Logger` log record by default.
```

Reviewer: I see why it'd make sense, but it'd be difficult to stay consistent over time - let's keep Powertools Logger. At least until we explore tools like [Vale](https://vale.sh/).

Author: Sounds good. I think this is ready to merge then.

> You can include any of these logging attributes as key value arguments (`kwargs`) when instantiating `Logger` or `LambdaPowertoolsFormatter`.

Reviewer: For example, here `Logger` is more explicit in intent than Logger.

You can also add them later anywhere in your code with `append_keys`, or remove them with `remove_keys` methods.

=== "collect.py"

    ```python hl_lines="3 8 10"
    ---8<-- "examples/logger/src/append_and_remove_keys.py"
    ```
=== "Example CloudWatch Logs excerpt"

    ```json hl_lines="6 15-16"
    ---8<-- "examples/logger/src/append_and_remove_keys.json"
    ```

> For log records originating from Powertools `Logger`, the `name` attribute will be the same as `service`, for log records coming from standard library logger, it will be the name of the logger (i.e. what was used as name argument to `logging.getLogger`).

Reviewer: ```suggestion
For log records originating from Powertools Logger, the `name` attribute will be the same as `service`, for log records coming from standard library logger, it will be the name of the logger (i.e. what was used as name argument to `logging.getLogger`).
```
